### PR TITLE
feat: add IdempotencyKey to prevent double payments

### PR DESCRIPTION
**Pull request by michelemendel**

### CODE REVIEW

**Comment by anthomm** [comment]
Please review [this](https://protobuf.dev/programming-guides/style/) style guide for protobuf.

File: proto/server/cutters_api.proto [hunk 1]
```diff
@@ -118,11 +118,13 @@ message PurchaseRequest {
   payments.PaymentRequest payment = 5;
   // If aware_of_campaigns is set, then the gateway will apply campaigns when calculating the price.
   bool aware_of_campaigns = 6;
+  string IdempotencyKey = 7;
```
Comment (anthomm) [inline proto/server/cutters_api.proto:121]: Please use the protocol buffer conventions for naming fields.

File: proto/server/cutters_api.proto [hunk 2]
```diff
@@ -118,11 +118,13 @@ message PurchaseRequest {
...
 }
 
 message PurchaseResponse {
   payments.PaymentResponse payment = 1;
   repeated BookingDetails bookings = 2;
+  string IdempotencyKey = 3;
```
Comment (anthomm) [inline proto/server/cutters_api.proto:127]: Please use the protocol buffer conventions for naming fields.

**Review by anthomm — CHANGES_REQUESTED** [review]

File: proto/server/cutters_api.proto (hunk 1 above)
Comment (michelemendel) [reply proto/server/cutters_api.proto]: fixed

File: proto/server/cutters_api.proto (hunk 2 above)
Comment (michelemendel) [reply proto/server/cutters_api.proto]: fixed

**Review by anthomm — APPROVED** [review]
LGTM

**Pull request merged**